src/main/scala/firrtl/ir/IR.scala

```diff
@@ -211,8 +211,29 @@ abstract class Width extends FirrtlNode {
   }
 }
 /** Positive Integer Bit Width of a [[GroundType]] */
-case class IntWidth(width: BigInt) extends Width {
+object IntWidth {
+  private val maxCached = 1024
+  private val cache = new Array[IntWidth](maxCached)
+  def apply(width: BigInt): IntWidth = {
+    if (1 <= width && width <= maxCached) {
+      val i = width.toInt - 1
+      Option(cache(i)) match {
```

Comment:
You should use null, not Option, as with scala's BigInt implementation.  Since the point of this optimization is to avoid object creation, making an Option is sort of self-defeating.